magicblock-committor-service/src/persist/commit_persister.rs

```diff
@@ -1,258 +1,574 @@
-                undelegate,
-                lamports: changed_account.lamports(),
-                finalize,
-                data,
-                commit_type,
-                created_at: now,
-                commit_status: CommitStatus::Pending,
-                last_retried_at: now,
-                retries_count: 0,
-            };
-
-            commit_rows.push(commit_row);
-        }
+    fn start_base_intent(
+        &self,
+        l1_message: &ScheduledBaseIntent,
+    ) -> CommitPersistResult<()> {
+        let commit_row = Self::create_commit_rows(l1_message);
+        self.commits_db
+            .lock()
+            .expect(POISONED_MUTEX_MSG)
+            .insert_commit_status_rows(&commit_row)?;
+
+        Ok(())
+    }
 
-        // Insert all commit rows into the database
-        self.db.insert_commit_status_rows(&commit_rows)?;
+    fn set_commit_id(
+        &self,
+        message_id: u64,
+        pubkey: &Pubkey,
+        commit_id: u64,
+    ) -> CommitPersistResult<()> {
+        self.commits_db
+            .lock()
+            .expect(POISONED_MUTEX_MSG)
+            .set_commit_id(message_id, pubkey, commit_id)
+    }
+
+    fn set_commit_strategy(
+        &self,
+        commit_id: u64,
+        pubkey: &Pubkey,
+        value: CommitStrategy,
+    ) -> CommitPersistResult<()> {
+        self.commits_db
+            .lock()
+            .expect(POISONED_MUTEX_MSG)
+            .set_commit_strategy(commit_id, pubkey, value)
+    }
 
-        Ok(reqid)
+    fn update_status_by_message(
+        &self,
+        message_id: u64,
+        pubkey: &Pubkey,
+        status: CommitStatus,
+    ) -> CommitPersistResult<()> {
+        self.commits_db
+            .lock()
+            .expect(POISONED_MUTEX_MSG)
+            .update_status_by_message(message_id, pubkey, &status)
     }
 
-    pub fn update_status(
-        &mut self,
-        reqid: &str,
+    fn update_status_by_commit(
+        &self,
+        commit_id: u64,
         pubkey: &Pubkey,
         status: CommitStatus,
-    ) -> Result<(), CommitPersistError> {
-        // NOTE: only Pending commits don't have a bundle id, but we should
-        //       never update to Pending
-        let Some(bundle_id) = status.bundle_id() else {
-            return Err(
-                CommitPersistError::CommitStatusUpdateRequiresStatusWithBundleId(
-                    status.as_str().to_string(),
-                ),
-            );
-        };
+    ) -> CommitPersistResult<()> {
+        self.commits_db
+            .lock()
+            .expect(POISONED_MUTEX_MSG)
+            .update_status_by_commit(commit_id, pubkey, &status)
+    }
 
-        let bundle_signature = status.signatures().map(|sigs| {
-            BundleSignatureRow::new(
-                bundle_id,
-                sigs.process_signature,
-                sigs.finalize_signature,
-                sigs.undelegate_signature,
-            )
-        });
+    fn get_commit_statuses_by_message(
```

Comment:
Not a huge fan of the adjustment to use `message` instead of `bundle`.
`bundle` expresses a lot better that something belongs together and has to be retried together (in a _bundled_ fashion) for instance.

Reply:
We don't have bundles, we have messages/intents they are never bundled with each other.